docs/src/userguide/extension_types.rst

```diff
@@ -484,6 +484,76 @@ statically sized freelist of ``N`` instances for a given type.  Example::
     penguin = Penguin('fish 2')  # does not need to allocate memory!
 
 
+.. _existing-pointers-instantiation:
+
+Instantiation from existing C/C++ pointers
+===========================================
+
+It is quite common to want to instantiate an extension class from an existing
+(pointer to a) data structure, often as returned by external C/C++ functions.
+
+As extension classes can only accept Python objects as arguments in their
+contructors, this necessitates the use of factory functions. For example, ::
+
+    from libc.stdlib cimport malloc, free
+
+    # Example C struct
+    ctypedef struct my_c_struct:
+        int a
+        int b
+
+
+    cdef class WrapperClass:
+        """A wrapper class for a C/C++ data structure"""
+        cdef my_c_struct *_ptr
+
+        def __cinit__(self):
+            # On cinit, do not create new structure but set pointer to NULL
+            self._ptr = NULL
+
+        def __dealloc__(self):
+            # De-allocate if we have a non-null pointer
```

Comment:
There should be discussion here about pointer ownership, possibly tracked by a boolean.

Reply:
Can you explain a little bit what you mean by pointer ownership? Like the class has a flag that if set, means it 'owns' the pointer and should de-allocate it?

Reply:
Yep.